slick.grid.js

```diff
@@ -1894,9 +1908,9 @@ if (typeof Slick === "undefined") {
       var ariaLabel = '';
       if (value) {
         if (typeof value === 'string') {
-          ariaLabel = `aria-label="${value}"`;
+          ariaLabel = `aria-label="${htmlEscape(value)}"`;
         } else if (value.ariaLabel) {
-          ariaLabel = `aria-label="${value.ariaLabel}"`;
+          ariaLabel = `aria-label="${htmlEscape(value.ariaLabel)}"`;
```

Comment:
What does the formatterResult look like? Do we not need to escape the actual value as well here?

Reply:
We should probably do ALL escaping in this library. Having it split across two separate places seems risky.

@alanrenmsft thoughts?

Reply:
I think what Alex had here is good enough, we can only handle the aria-label here since the cells could have different formatters: could be text, link, button or other types.

Reply:
But the act of escaping HTML should never be relegated to the calling library - that needs to be something that is controlled here IMO. This has two benefits : 

1. Ensures that the content is ALWAYS escaped (guarantees this issue doesn't happen again if we use this elsewhere)
2. Ensures the escaping logic is the same so that any bugs only have to be fixed in one place

Reply:
what you get from the formatter is a string, it could be plain text or html, not sure how easy it is to find out the attribute values in the html to be escaped.

Reply:
What do you mean the "attribute values"? The value is directly inserted into the HTML - so you just escape all the HTML characters (using the function Alex brought over above)

Reply:
the value could be `<button aria-label="xxx">Delete</button>`, we should only escape the 'Delete' and 'xxx', we can't escape the whole string.

Reply:
I see...I didn't realize we're passing in HTML that's intended to be rendered. That's an unfortunate pattern but I guess it's too late to do much about that now.